pkg/node/fsm/ng_state.go

```diff
@@ -354,8 +423,22 @@ func (c *blockStatesCache) Get(blockID proto.BlockID) (*proto.Block, bool) {
 	return &block, true
 }
 
+func (c *blockStatesCache) GetSnapshot(blockID proto.BlockID) (*proto.BlockSnapshot, bool) {
+	snapshot, ok := c.snapshots[blockID]
+	if !ok {
+		return nil, false
+	}
+	return &snapshot, true
+}
+
 func initNGStateInFSM(state *StateData, fsm *stateless.StateMachine, info BaseInfo) {
 	var ngSkipMessageList proto.PeerMessageIDs
+	if !info.enableLightMode {
+		ngSkipMessageList = append(ngSkipMessageList, []proto.PeerMessageID{
+			proto.ContentIDMicroBlockSnapshot,
+			proto.ContentIDBlockSnapshot,
+		}...)
+	}
```

Comment:
These messages should be ignored in `ng` mode handled only inside `WaitMicroSnapshotState` and `WaitSnapshotState`.